core/graph/src/main/java/uk/gov/gchq/gaffer/graph/Graph.java

```diff
@@ -854,199 +862,151 @@ public Graph build() {
                 } else {
-                    schema = new Schema.Builder()
-                            .merge(mergedParentSchema)
-                            .merge(schema)
-                            .build();
-                }
-            }
-
-            if (!schemaBytesList.isEmpty()) {
-                if (null == properties) {
-                    throw new IllegalArgumentException("To load a schema from json, the store properties must be provided.");
-                }
-
-                final Class<? extends Schema> schemaClass = properties.getSchemaClass();
-                final Schema newSchema = new Schema.Builder()
-                        .json(schemaClass, schemaBytesList.toArray(new byte[schemaBytesList.size()][]))
+                    mergedSchema = new Schema.Builder()
+                        .merge(mergedSchema)
+                        .merge(parentSchema)
                         .build();
-                addSchema(newSchema);
+                }
             }
-        }
-
-        private void updateStoreProperties(final GraphConfig config) {
-            StoreProperties mergedStoreProperties = null;
-            if (null != parentStorePropertiesId) {
-                mergedStoreProperties = config.getLibrary().getProperties(parentStorePropertiesId);
+            // Merge parent schemas with current schema
+            if (mergedSchema != null && currentSchema != null) {
+                mergedSchema = new Schema.Builder()
+                    .merge(mergedSchema)
+                    .merge(currentSchema)
+                    .build();
             }
 
-            if (null != properties) {
-                if (null == mergedStoreProperties) {
-                    mergedStoreProperties = properties;
-                } else {
-                    mergedStoreProperties.merge(properties);
-                }
-            }
-            properties = mergedStoreProperties;
+            return mergedSchema;
         }
 
-        private void updateStore(final GraphConfig config) {
-            if (null == store) {
+        /**
+         * Initialises the {@link Store} class instance with a given graph configuration.
+         * <p>
+         * Will also attempt to extract some configuration from the current store if
+         * not currently configured.
+         *
+         * @param config The graph config
+         */
+        private void initStore(final GraphConfig config) {
+            // Init the store if needed
+            if (store == null) {
+                LOGGER.debug("Store currently null initialising with Id: {} and existing schema/properties", config.getGraphId());
                 store = Store.createStore(config.getGraphId(), cloneSchema(schema), properties);
-            } else if ((null != config.getGraphId() && !config.getGraphId().equals(store.getGraphId()))
-                    || (null != schema)
-                    || (null != properties && !properties.equals(store.getProperties()))) {
```

Comment:
This logic has changed, so that these `if`s are now run unconditionally?

Reply:
The original logic didn't really make much sense technically not much has really changed though. 

Before to get into this block at least one of those conditions must've been met but then you don't do anything unless the value is `null`. Take for example the `schema`, to get into this block it either needs to be set or one of the other values are set at which point you don't do anything with it anyways as the next check is to make sure it is `null`. Therefore, conditionally the outer block is basically making no difference as to what gets called internally so may as well remove it.

Reply:
Ah ok, just wanted to be sure.